lib/usecases/use_case_result.dart

```diff
@@ -1,22 +1,24 @@
 part of 'base_use_case.dart';
 
-abstract class Result<T> {}
+abstract class Result<T> {
+  Result._();
+}
 
 class Success<T> extends Result<T> {
   final T value;
 
-  Success(this.value);
+  Success(this.value) : super._();
 }
 
-class ApiError implements Exception {
+class UseCaseError implements Exception {
```

Comment:
What do you think about rename `UserCaseError` to `UserCaseException`?
Refer from [Error](https://api.dart.dev/stable/2.12.2/dart-core/Error-class.html) & [Exception](https://api.dart.dev/stable/2.12.2/dart-core/Exception-class.html)

Reply:
for this renaming it makes sense. Subclass must extends the superclass name 🤙